### PR TITLE
[T3A3][T09-A3] Tang Di Feng

### PR DESCRIPTION
**Pull request by e0011840**

### CODE REVIEW

File: doc/UserGuide.md [hunk 1]
```diff
@@ -45,7 +45,7 @@ Format: `add NAME [p]p/PHONE_NUMBER [p]e/EMAIL [p]a/ADDRESS [t/TAG]...`
 > Persons can have any number of tags (including 0)
 
 Examples: 
-* `add John Doe p/98765432 e/johnd@gmail.com a/John street, block 123, #01-01`
+* `add John Doe p/98765432 e/johnd@gmail.com a/John street, block 123, #01-01, 123456`
```
Comment (okkhoy) [inline doc/UserGuide.md]: user guide should also indicate the format of the address now supported explicitly; not just an example

File: src/seedu/addressbook/commands/AddCommand.java [hunk 2]
```diff
@@ -19,7 +19,7 @@
             + "Contact details can be marked private by prepending 'p' to the prefix.\n"
             + "Parameters: NAME [p]p/PHONE [p]e/EMAIL [p]a/ADDRESS  [t/TAG]...\n"
             + "Example: " + COMMAND_WORD
-            + " John Doe p/98765432 e/johnd@gmail.com a/311, Clementi Ave 2, #02-25 t/friends t/owesMoney";
+            + " John Doe p/98765432 e/johnd@gmail.com a/311, Clementi Ave 2, #02-25, 233242 t/friends t/owesMoney";
```
Comment (okkhoy) [inline src/seedu/addressbook/commands/AddCommand.java]: :+1: good to see you have updated the in app help :-)

File: src/seedu/addressbook/data/person/Address.java [hunk 3]
```diff
@@ -9,8 +13,14 @@
 public class Address {
 
     public static final String EXAMPLE = "123, some street";
-    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
+    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses MUST be in 'a/123, Clementi Ave 3, #12-34, 231534'.";
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Address.java]: hmmmm, how are you enforcing this format? how are you checking if the address input is valid?

File: src/seedu/addressbook/data/person/Address.java [hunk 4]
```diff
@@ -27,6 +37,12 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
         this.value = trimmedAddress;
+        String[] splitaddress = trimmedAddress.split(DELIMITER);
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Address.java]: good! now you can use any character as delimiter!

File: src/seedu/addressbook/data/person/Address.java [hunk 5]
```diff
@@ -27,6 +37,12 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
...
+        this.block = new Block(splitaddress[0]);
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Address.java]: why not 1? or 2? or 100?
I hope you understand what I mean! ;-) may be i revoke the good from the previous comment :-P

File: src/seedu/addressbook/data/person/Block.java [hunk 6]
```diff
@@ -0,0 +1,13 @@
+package seedu.addressbook.data.person;
+
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Block.java]: missing header comments. how would users / developers know what does `Block` mean?

File: test/input.txt [hunk 7]
```diff
@@ -40,19 +40,19 @@
   add Valid Name p/12345 e/valid@e.mail a/valid, address t/invalid_-[.tag
 
   # should add correctly and list non private information
-  add Adam Brown p/111111 e/adam@gmail.com a/111, alpha street
+  add Adam Brown p/111111 e/adam@gmail.com a/111, alpha street, #06-111, 234523
```
Comment (okkhoy) [inline test/input.txt]: you are not testing for invalid address, are you?

**Comment by okkhoy** [comment]
@e0011840 some comments added, please check.
please close the PR once you have noted the suggestions.

**Comment by e0011840** [comment]
Noted With thanks!!

**Pull request closed without merging**